### PR TITLE
Plant packaging prebuilt Linux GPU miners

### PR DESCRIPTION
**Pull request by lucafavatella**
Delivers https://www.pivotaltracker.com/story/show/162198878

TODO
* [x] Determine reliable way for extracting files from git repo. `pax` (for processing `git archive` output) is POSIX but apparently sometimes absent, [at least on CI](https://circleci.com/gh/aeternity/epoch/47838)
* [x] Split in separate repos:
  * [x] [Repo meant to contain prebuilt miners, as repo `aeternity/cuckoo-prebuilt`](https://github.com/aeternity/epoch/commits/PT-162198878-package-gpu-app--miner-binaries-repo)
    * https://github.com/aeternity/cuckoo-prebuilt
  * [ ] ~[Repo meant to contain OTP app including the prebuilt miners](https://github.com/aeternity/epoch/commits/PT-162198878-package-gpu-app--otp-app-repo)~
  * [x] [Repo meant to contain rebar3 plugin generating OTP app including the prebuilt miners](https://github.com/aeternity/epoch/commits/PT-162198878-package-gpu-app--rebar3-plugin-repo)
    * https://github.com/aeternity/rebar3-cuckoo-prebuilt-plugin
* [ ] ~Review *ignore files~
* [ ] ~Reduce noise in repeated invocations of Makefile~
* [ ] ~Make Makefile more resilient to existing repo in c_src~
* [x] Prevent `.git` directory of prebuilt miners repo from being included in OTP release
* [x] Add actual Linux GPU miner to repo meant to contain prebuilt miners
  * [x] Add miner to https://github.com/aeternity/cuckoo-prebuilt
    * [Done](https://github.com/aeternity/cuckoo-prebuilt/commit/54f098d45a25490cd22bc80c87c7dcc937d53382). Built semi-automatically using https://github.com/aeternity/cuckoo/pull/11
* [x] Add release note line
* [x] Mention in docs
* [ ] ~Fix system test job re cache~
  * ~Maybe bump cache version~
  * Not needed any longer somehow

Maybe TODO:
* [x] Include miners only for OS for which package being built
* [ ] Replace miners in aecuckoo app

### CODE REVIEW

**Comment by lucafavatella** [comment]
[CI job running full system tests](https://circleci.com/gh/aeternity/epoch/48319). I disabled cache in that job as I noticed some cache-related failures.

**Comment by dincho** [comment]
> [CI job running full system tests](https://circleci.com/gh/aeternity/epoch/48319). I disabled cache in that job as I noticed some cache-related failures.
> 
> EDIT: Passed.

You can bump the cache version to invalidate it

File: apps/aecore/src/aecore.app.src [hunk 1]
```diff
@@ -34,6 +34,7 @@
     unicode_util_compat, %% to be removed, not needed in OTP 20
     idna,
     aecuckoo,
```
Comment (dincho) [inline apps/aecore/src/aecore.app.src]: Why do we keep the old miners ?

Comment (lucafavatella) [reply apps/aecore/src/aecore.app.src]: Yes.

This enables user to configure prebuilt miner while deferring solving the problem of identifying which OS needs to be selected.

File: rebar.config [hunk 2]
```diff
@@ -41,7 +41,7 @@
         {sha3, {git, "https://github.com/szktty/erlang-sha3", {ref, "dbdfd12"}}},
         {sext, {git, "https://github.com/uwiger/sext.git", {ref, "615eebc"}}},
         {idna, {git, "https://github.com/aeternity/erlang-idna", {tag, "6.0.0"}}},
-        {aecuckooprebuilt, {aecuckooprebuilt_app_with_priv_from_git, {git, "https://github.com/aeternity/cuckoo-prebuilt.git", {ref, "54f098d45a25490cd22bc80c87c7dcc937d53382"}}}}
+        {aecuckooprebuilt, {aecuckooprebuilt_app_with_priv_from_git, {git, "https://github.com/aeternity/cuckoo-prebuilt.git", {ref, "90afb699dc9cc41d033a7c8551179d32b3bd569d"}}}}
```
Comment (lucafavatella) [inline rebar.config]: Using artefact from [CI](https://circleci.com/gh/aeternity/epoch/48689#artifacts/containers/0) on this commit:
```
$ tar tf ~/Downloads/epoch-932ca2c09dd14bf5e48bba666ea1c64bc072b467-ubuntu-x86_64.tar.gz  | grep prebuilt
lib/aecuckooprebuilt-0.1.0/priv/cuda29
lib/aecuckooprebuilt-0.1.0/priv/lcuda29
lib/aecuckooprebuilt-0.1.0/ebin/aecuckooprebuilt.app
```

**Comment by lucafavatella** [comment]
Rebased on recent master, and force-pushed.

> executable_group should be documented in cuda-miner.md

Done.

**Comment by lucafavatella** [comment]
Link to package built on 23cc210 https://48972-99802036-gh.circle-artifacts.com/0/epoch-23cc2105c0f400ec550c0c88d56b0a97a47d0104-ubuntu-x86_64.tar.gz

**Pull request merged**